README.md

```diff
@@ -1,21 +1,56 @@
 # Workflow Status Slack Notification Action
 
-> Github action that keeps track and retrieves previous workflow runs status for the same workflow on the same branch. 
-  According to current and previous runs status, it will send a notification to a Slack channel using an Incoming webhook app.
+> Github action that keeps track and retrieves previous workflow runs status for the same workflow on the same branch.
+> According to current and previous runs status, it will send a notification to a Slack channel using an Incoming webhook app.
 
 ## How it works
 
 At the end of a workflow, on success() and on failure(), call this action with the current status as a parameter (success/failure).
-This action will first check the previous status and according to the current status. It will send a notification to Slack if:
+This action will first check the previous workflow status and will send a Slack notification according to the current workflow status.
+Status is retrieved from cache if the current workflow has already run previously (from `re-run all jobs` button for example).
+If it's the first time this workflow runs, this action will retrieve previous workflow status on the same branch with `gh` cli using provided `github-token`.
```

Comment:
Why will we retrieve a previous workflow state on the first try?

Reply:
We can have multiple workflows per branch/PR, and multiple runs per workflow.
If it's the first workflow run, we check if there has been another workflow before for the specific branch/PR.
Let me know if it's still not clear